client/src/features/home/Home.tsx

```diff
@@ -161,6 +163,25 @@ function Home(): JSX.Element {
   const classes = useStyles();
   const [selectedSearchCategory, setSelectedSearchCategory] = React.useState<string>('');
   const [searchText, setSearchText] = React.useState<string>('');
+  const [donations, setDonations] = React.useState<Asset[]>([]);
```

Comment:
Recommended:
Can you use the javascript URL api to generate the url?
https://www.valentinog.com/blog/url/